### PR TITLE
remove google java formatter dependency

### PR DESCRIPTION
**Pull request by jsvd**

### CODE REVIEW

**Comment by jsvd** [comment]
@colinsurprenant you've been around the compilation code recently, what do you think here about the need of this dependency? if this is being used to avoid cache invalidation, in what scenarios does it do that?

This dependency was introduced in https://github.com/elastic/logstash/pull/8913/

**Comment by colinsurprenant** [comment]
Good question, will investigate.

**Comment by colinsurprenant** [comment]
@jsvd  This is used to do generated source code formatting and in that respect insures that 2 equivalent code generation will yield the same text+formatting. I see that the note about licensing :

> WARNING: DO NOT UPGRADE "google-java-format"
> later versions require GPL licensed code in javac-shaded that is
> Apache2 incompatible

I just verified that and starting at v1.8 they removed that GPL dependency  we should be good to upgrade to [latest v1.9 ](https://mvnrepository.com/artifact/com.google.googlejavaformat/google-java-format/1.9) without incompatible license dependency. 

I think we can close this and create a new PR to upgrade google-java-format.

**Comment by colinsurprenant** [comment]
Created #12223 to upgrade to latest versions.

**Comment by jsvd** [comment]
@colinsurprenant I believe that version only works with Java >= 9, which is why we didn’t upgrade

**Comment by colinsurprenant** [comment]
Good catch. I can't find clear indications for the Java version requirements but looking at https://github.com/google/google-java-format/releases all I see is that v1.8 requires JDK 11. 

Short of upgrading the library because of minimum JDK requirements we can either keep the current version or remove it altogether as you suggest here but by doing so we also will loose readable formatting on the generated sources which is useful when developing/debugging. I don't think that removing the formatter will break caching but I don't know for sure.

**Comment by andsel** [comment]
Here we should upgrade to [version 1.13](https://mvnrepository.com/artifact/com.google.googlejavaformat/google-java-format/1.13.0).

File: logstash-core/build.gradle [hunk 1]
```diff
@@ -183,12 +183,6 @@ dependencies {
         api files(customJRubyDir + "/maven/jruby-complete/target/jruby-complete-${customJRubyVersion}.jar")
     }
     implementation group: 'com.google.guava', name: 'guava', version: '24.1.1-jre'
-    // WARNING: DO NOT UPGRADE "google-java-format"
-    // later versions require GPL licensed code in javac-shaded that is
-    // Apache2 incompatible
-    implementation('com.google.googlejavaformat:google-java-format:1.1') {
```
Comment (andsel) [inline logstash-core/build.gradle:189]: Given that it's use for code formatting in debug mode of Janino, we should keep it and update to version 1.13 which still require Java 11 and is In Apache License

**Review by andsel — CHANGES_REQUESTED** [review]
Left a comment to update the version

**Comment by jsvd** [comment]
Closing this in favor of https://github.com/elastic/logstash/pull/13700

**Pull request closed without merging**